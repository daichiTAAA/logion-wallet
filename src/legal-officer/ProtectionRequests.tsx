import React from 'react';

import { ContentPane } from '../component/Dashboard';
import Frame from '../component/Frame';
import { useRootContext } from '../RootContext';

import { useLegalOfficerContext } from './LegalOfficerContext';
import PendingProtectionRequests from './PendingProtectionRequests';
import ProtectionRequestsHistory from './ProtectionRequestsHistory';
import RefreshTokenizationRequestsButton from './RefreshTokenizationRequestsButton';
import ProtectedUsers from "./ProtectedUsers";

export default function ProtectionRequests() {
    const { selectAddress, addresses } = useRootContext();
    const { colorTheme } = useLegalOfficerContext();

    if(addresses === null || selectAddress === null) {
        return null;
    }

    return (
        <ContentPane
            mainTitle="Protection requests"
            colors={ colorTheme }
            addresses={ addresses }
            selectAddress={ selectAddress }
            primaryAreaChildren={
                <>
                    <Frame
                            colors={colorTheme}
                    >
                        <RefreshTokenizationRequestsButton/>
                        <PendingProtectionRequests recovery={ false } />
                    </Frame>
                    <h1>Activated User Account Protection under my watch</h1>
                    <Frame
                            colors={colorTheme}
                    >
                        <ProtectedUsers/>
                    </Frame>
                </>
            }
            secondaryAreaChildren={
                <Frame
                    colors={ colorTheme }
                >
                    <ProtectionRequestsHistory recovery={ false } />
                </Frame>
            }
        />
    );
}
