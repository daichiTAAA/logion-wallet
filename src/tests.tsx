import { ReactElement } from 'react';
import { createRenderer } from 'react-test-renderer/shallow';
import reactRenderer, { ReactTestRenderer, ReactTestRendererJSON, act as reactAct } from 'react-test-renderer';

export function render(element: ReactElement): null | ReactTestRendererJSON | ReactTestRendererJSON[] {
    const component = reactRenderer.create(element);
    return component.toJSON();
}

export function renderer(element: ReactElement): ReactTestRenderer {
    return reactRenderer.create(element);
}

export function shallowRender<E extends ReactElement>(element: ReactElement): E {
    const renderer = createRenderer();
    renderer.render(element);
    return renderer.getRenderOutput();
}

export function mockAccount(address: string, name: string) {
    return {
        address,
        meta: {
            name
        }
    };
}

export { reactAct as act };
