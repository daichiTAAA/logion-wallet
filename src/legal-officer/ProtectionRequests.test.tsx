jest.mock("../common/CommonContext");

import React from 'react';
import ProtectionRequests from './ProtectionRequests';
import { shallowRender } from '../tests';

test('renders', () => {
    const tree = shallowRender(<ProtectionRequests />);
    expect(tree).toMatchSnapshot();
});
