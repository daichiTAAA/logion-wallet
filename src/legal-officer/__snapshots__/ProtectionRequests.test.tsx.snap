// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<ContentPane
  addresses={
    Object {
      "addresses": Array [
        Object {
          "address": "5H4MvAsobfZ6bBCDyj5dsrWYLrA8HrRzaqa9p61UXtxMhSCY",
          "isLegalOfficer": false,
          "name": "name",
        },
      ],
      "currentAddress": Object {
        "address": "5H4MvAsobfZ6bBCDyj5dsrWYLrA8HrRzaqa9p61UXtxMhSCY",
        "isLegalOfficer": false,
        "name": "name",
      },
    }
  }
  colors={
    Object {
      "accounts": Object {
        "background": "eff3fe",
        "foreground": "#000000",
        "hintColor": "#00000099",
        "iconBackground": "#3b6cf4",
        "legalOfficerIcon": Object {
          "category": "legal-officer",
          "id": "account-shield",
        },
      },
      "bottomMenuItems": Object {
        "iconGradient": Object {
          "from": "#7a90cb",
          "to": "#3b6cf4",
        },
      },
      "buttons": Object {
        "secondaryBackgroundColor": "#eff3fe",
      },
      "dashboard": Object {
        "background": "#eff3fe",
        "foreground": "#000000",
      },
      "frame": Object {
        "background": "#ffffff",
        "foreground": "#000000",
        "link": "#3b6cf426",
      },
      "select": Object {
        "background": "#d8e2fd",
        "foreground": "#000000",
        "menuBackgroundColor": "#d8e2fd",
        "selectedOptionBackgroundColor": "#ffffff33",
      },
      "shadowColor": "#3b6cf41a",
      "sidebar": Object {
        "activeItemBackground": "#d8e2fd",
        "background": "#ffffff",
        "foreground": "#000000",
      },
      "topMenuItems": Object {
        "iconGradient": Object {
          "from": "#3b6cf4",
          "to": "#6050dc",
        },
      },
      "type": "light",
    }
  }
  mainTitle="Protection requests"
  primaryAreaChildren={
    <React.Fragment>
      <Frame
        colors={
          Object {
            "accounts": Object {
              "background": "eff3fe",
              "foreground": "#000000",
              "hintColor": "#00000099",
              "iconBackground": "#3b6cf4",
              "legalOfficerIcon": Object {
                "category": "legal-officer",
                "id": "account-shield",
              },
            },
            "bottomMenuItems": Object {
              "iconGradient": Object {
                "from": "#7a90cb",
                "to": "#3b6cf4",
              },
            },
            "buttons": Object {
              "secondaryBackgroundColor": "#eff3fe",
            },
            "dashboard": Object {
              "background": "#eff3fe",
              "foreground": "#000000",
            },
            "frame": Object {
              "background": "#ffffff",
              "foreground": "#000000",
              "link": "#3b6cf426",
            },
            "select": Object {
              "background": "#d8e2fd",
              "foreground": "#000000",
              "menuBackgroundColor": "#d8e2fd",
              "selectedOptionBackgroundColor": "#ffffff33",
            },
            "shadowColor": "#3b6cf41a",
            "sidebar": Object {
              "activeItemBackground": "#d8e2fd",
              "background": "#ffffff",
              "foreground": "#000000",
            },
            "topMenuItems": Object {
              "iconGradient": Object {
                "from": "#3b6cf4",
                "to": "#6050dc",
              },
            },
            "type": "light",
          }
        }
      >
        <RefreshTokenizationRequestsButton />
        <PendingProtectionRequests
          recovery={false}
        />
      </Frame>
      <h1>
        Activated User Account Protection under my watch
      </h1>
      <Frame
        colors={
          Object {
            "accounts": Object {
              "background": "eff3fe",
              "foreground": "#000000",
              "hintColor": "#00000099",
              "iconBackground": "#3b6cf4",
              "legalOfficerIcon": Object {
                "category": "legal-officer",
                "id": "account-shield",
              },
            },
            "bottomMenuItems": Object {
              "iconGradient": Object {
                "from": "#7a90cb",
                "to": "#3b6cf4",
              },
            },
            "buttons": Object {
              "secondaryBackgroundColor": "#eff3fe",
            },
            "dashboard": Object {
              "background": "#eff3fe",
              "foreground": "#000000",
            },
            "frame": Object {
              "background": "#ffffff",
              "foreground": "#000000",
              "link": "#3b6cf426",
            },
            "select": Object {
              "background": "#d8e2fd",
              "foreground": "#000000",
              "menuBackgroundColor": "#d8e2fd",
              "selectedOptionBackgroundColor": "#ffffff33",
            },
            "shadowColor": "#3b6cf41a",
            "sidebar": Object {
              "activeItemBackground": "#d8e2fd",
              "background": "#ffffff",
              "foreground": "#000000",
            },
            "topMenuItems": Object {
              "iconGradient": Object {
                "from": "#3b6cf4",
                "to": "#6050dc",
              },
            },
            "type": "light",
          }
        }
      >
        <ProtectedUsers />
      </Frame>
    </React.Fragment>
  }
  secondaryAreaChildren={
    <Frame
      colors={
        Object {
          "accounts": Object {
            "background": "eff3fe",
            "foreground": "#000000",
            "hintColor": "#00000099",
            "iconBackground": "#3b6cf4",
            "legalOfficerIcon": Object {
              "category": "legal-officer",
              "id": "account-shield",
            },
          },
          "bottomMenuItems": Object {
            "iconGradient": Object {
              "from": "#7a90cb",
              "to": "#3b6cf4",
            },
          },
          "buttons": Object {
            "secondaryBackgroundColor": "#eff3fe",
          },
          "dashboard": Object {
            "background": "#eff3fe",
            "foreground": "#000000",
          },
          "frame": Object {
            "background": "#ffffff",
            "foreground": "#000000",
            "link": "#3b6cf426",
          },
          "select": Object {
            "background": "#d8e2fd",
            "foreground": "#000000",
            "menuBackgroundColor": "#d8e2fd",
            "selectedOptionBackgroundColor": "#ffffff33",
          },
          "shadowColor": "#3b6cf41a",
          "sidebar": Object {
            "activeItemBackground": "#d8e2fd",
            "background": "#ffffff",
            "foreground": "#000000",
          },
          "topMenuItems": Object {
            "iconGradient": Object {
              "from": "#3b6cf4",
              "to": "#6050dc",
            },
          },
          "type": "light",
        }
      }
    >
      <ProtectionRequestsHistory
        recovery={false}
      />
    </Frame>
  }
  selectAddress={[MockFunction]}
/>
`;
